docs/usage/auth_methods/jwt-oidc.rst

```diff
@@ -142,36 +142,76 @@ OIDC Authorization URL Request
 
 :py:meth:`hvac.api.auth_methods.JWT.oidc_authorization_url_request`
 
+This is equivalent to `vault login -method=oidc`
+
 .. code:: python
 
     import webbrowser
     import http.server
     import hvac
-    client = hvac.Client()
-
-    auth_url_response = client.auth.oidc.oidc_authorization_url_request(
-        role='hvac',
-        redirect_uri='http://localhost:8250/oidc/callback'
-    )
-    auth_url = auth_url_response['data']['auth_url']
-    if auth_url == '':
-        return None
-
-    params = parse.parse_qs(auth_url.split('?')[1])
-    auth_url_nonce = params['nonce'][0]
-    auth_url_state = params['state'][0]
-
-    webbrowser.open(auth_url)
-    token = login_odic_get_token()
-
-    auth_result = client.auth.oidc.oidc_callback(
-        code=token, path='oidc', nonce=auth_url_nonce, state=auth_url_state
-    )
-
-    print('Client token returned: %s' % auth_result['auth']['client_token'])
+    import urllib.parse
+
+    # CHANGEME: these params might have to be changed to match your Vault configuration.
+    # Specifically
+    # 1. auth/oidc/role/XXX allowed_redirect_uris must contain the
+    #    REDIRECT_URI string used below.
+    # 2. Role must match your environment's role for this client.
+    PORT = 8250
+    REDIRECT_URI = f'http://localhost:{PORT}/oidc/callback'
+    ROLE = 'hvac' # Use None (not empty string) for the default Role
+    SELF_CLOSING_PAGE = '''
+    <!doctype html>
+    <html>
+    <head>
+    <script>
+    // Closes IE, Edge, Chrome, Brave
+    window.onload = function load() {
+      window.open('', '_self', '');
+      window.close();
+    };
+    </script>
+    </head>
+    <body>
+      <p>Authentication successful, you can close the browser now.</p>
+      <script>
+        // Needed for Firefox security
+        setTimeout(function() {
+              window.close()
+        }, 5000);
+      </script>
+    </body>
+    </html>
+    '''
+
+    def main():
+        client = hvac.Client()
+
+        auth_url_response = client.auth.oidc.oidc_authorization_url_request(
+            role=ROLE,
+            redirect_uri=REDIRECT_URI,
+        )
+        auth_url = auth_url_response['data']['auth_url']
+        if auth_url == '':
+            return None # TODO: throw a nicer error
+
+        params = urllib.parse.parse_qs(auth_url.split('?')[1])
+        auth_url_nonce = params['nonce'][0]
+        auth_url_state = params['state'][0]
+
+        webbrowser.open(auth_url)
+        token = login_oidc_get_token()
+
+        auth_result = client.auth.oidc.oidc_callback(
+            code=token,
+            path='oidc',
+            nonce=auth_url_nonce,
+            state=auth_url_state,
+        )
+
+        print('Client token returned: %s' % auth_result['auth']['client_token'])
```

Comment:
Since we're updating this example, and using an f-string already on line 160, what do you think about updating this line to use an f-string as well?

Reply:
Fixed to f-string.